react-app/src/modules/account/account.js

```diff
@@ -0,0 +1,98 @@
+import scrypt from 'scrypt-async';
+import nacl from 'tweetnacl';
+import sha256 from 'fast-sha256';
+import blake from 'blakejs';
+
+nacl.util = require('tweetnacl-util');
+
+let _username, _pk, _sk, _tz, _offset;
+
+export function clearAccount() {
+  _username = _pk = _sk = _tz = _offset = undefined;
+}
+
+export function _setKeyPair(pk, sk) {
+  _pk = nacl.util.decodeBase64(pk);
+  _sk = nacl.util.decodeBase64(sk);
+}
+
+function _generateNonce(publicKeyA, publicKeyB) {
+  let state = blake.blake2bInit(nacl.box.nonceLength, null);
+  blake.blake2bUpdate(state, publicKeyA);
+  blake.blake2bUpdate(state, publicKeyB);
+  return blake.blake2bFinal(state);
+}
+
+export function constructor(username, password) {
+  return new Promise(resolve => {
+    scrypt(sha256(nacl.util.decodeUTF8(password)), sha256(nacl.util.decodeUTF8(username)), {
```

Comment:
This is a matter of personal taste, so you are not required to change it.

But I find the code more readable if I extract the complexity outside a function call and use variable names to document its purpose.

I'd write this function as follows:

```js
export function constructor(username, password) {
  return new Promise(resolve => {
    const hashedPassword = sha256(nacl.util.decodeUTF8(password));
    const salt = sha256(nacl.util.decodeUTF8(username));
    const options = {
      N: 1 << 20,
      r: 8,
      p: 1,
      dkLen: 32,
      encoding: 'binary'
    };

    const callback = (derivedKey) => {
      const keyPair = nacl.box.keyPair.fromSecretKey(derivedKey);
      _username = username;
      _pk = keyPair.publicKey;
      _sk = keyPair.secretKey;
      resolve(null);
    };

    scrypt(hashedPassword, salt, options, callback);
  });
}
```

What do you think?

Reply:
I will take this and the other recommendations to improve the readability of the code (they actually makes the code truly clear). I have some bad habits to get ride off and it's great to get this good practices tips to improve code =)